cluster/gce/gci/configure-helper.sh

```diff
@@ -1729,21 +1793,18 @@ function start-kube-apiserver {
     params+=" --kubelet-certificate-authority=${CA_CERT_BUNDLE_PATH}"
   fi
 
-  local admission_controller_config_mount=""
-  local admission_controller_config_volume=""
-  local image_policy_webhook_config_mount=""
-  local image_policy_webhook_config_volume=""
   if [[ -n "${ADMISSION_CONTROL:-}" ]]; then
     params+=" --admission-control=${ADMISSION_CONTROL}"
-    if [[ ${ADMISSION_CONTROL} == *"ImagePolicyWebhook"* ]]; then
-      params+=" --admission-control-config-file=/etc/admission_controller.config"
-      # Mount the file to configure admission controllers if ImagePolicyWebhook is set.
-      admission_controller_config_mount="{\"name\": \"admissioncontrollerconfigmount\",\"mountPath\": \"/etc/admission_controller.config\", \"readOnly\": false},"
-      admission_controller_config_volume="{\"name\": \"admissioncontrollerconfigmount\",\"hostPath\": {\"path\": \"/etc/admission_controller.config\", \"type\": \"FileOrCreate\"}},"
-      # Mount the file to configure the ImagePolicyWebhook's webhook.
-      image_policy_webhook_config_mount="{\"name\": \"imagepolicywebhookconfigmount\",\"mountPath\": \"/etc/gcp_image_review.config\", \"readOnly\": false},"
-      image_policy_webhook_config_volume="{\"name\": \"imagepolicywebhookconfigmount\",\"hostPath\": {\"path\": \"/etc/gcp_image_review.config\", \"type\": \"FileOrCreate\"}},"
-    fi
+    params+=" --admission-control-config-file=/etc/srv/kubernetes/admission_controller_config.yaml"
+  fi
+
+  # If GKE exec auth support is requested for webhooks, then
+  # gke-exec-auth-plugin needs to be mounted into the kube-apiserver container.
+  local webhook_exec_auth_plugin_mount=""
+  local webhook_exec_auth_plugin_volume=""
+  if [[ ${WEBHOOK_GKE_EXEC_AUTH:-} ]]; then
```

Comment:
It is a bit concerning that the logic for exec webhook are scattered among a couple places. Is there a specific reason the mount/volume setup has to be placed here?

Reply:
Here I was mostly following the existing organization of the file.

Note that this block isn't doing mounts *for* use by gke-exec-auth-plugin, rather it's mounting gke-exec-auth-plugin *into* the kube-apiserver container (so that the various webhooks can execute it).  To do so, it needs to modify the kube-apiserver's static pod manifest, which is what this function is all about.

Reply:
Makes sense, thanks for the explanation :)